base/base/find_symbols.h

```diff
@@ -211,6 +362,11 @@ inline char * find_first_symbols(char * begin, char * end)
     return const_cast<char *>(detail::find_first_symbols_dispatch<true, detail::ReturnMode::End, symbols...>(begin, end));
 }
 
+inline const char * find_first_symbols(std::string_view haystack, const SearchSymbols & symbols)
```

Comment:
Consider using methods from `std::`, please, if possible.
Are these functions introduced only for performance reasons?

Reply:
Yes, for performance reasons. According my "silly" benchmarks, SSE4 presents better performance when keys and values are bigger.

Plus, these functions already exist in a compile-time version. Do you think we can keep it?

Reply:
According [to a benchmark](https://quick-bench.com/#eyJ0ZXh0IjoiI3ByYWdtYSBjbGFuZyBhdHRyaWJ1dGUgcHVzaCAoX19hdHRyaWJ1dGVfXygodGFyZ2V0KFwic3NlLHNzZTIsc3NlNC4yXCIpKSksIGFwcGx5X3RvPWZ1bmN0aW9uKVxuXG4jaW5jbHVkZSA8Y3N0ZGludD5cbiNpbmNsdWRlIDxzdHJpbmc+XG4jaW5jbHVkZSA8c3RyaW5nX3ZpZXc+XG4jaW5jbHVkZSA8c3RyaW5nLmg+XG5cbiNpbmNsdWRlIDxhbGdvcml0aG0+XG4jaW5jbHVkZSA8ZXhlY3V0aW9uPlxuI2luY2x1ZGUgPHR5cGVfdHJhaXRzPlxuXG5cbi8vI2lmIGRlZmluZWQoX19TU0UyX18pXG4gICAgI2luY2x1ZGUgPGVtbWludHJpbi5oPlxuLy8jZW5kaWZcbi8vI2lmIGRlZmluZWQoX19TU0U0XzJfXylcbiAgICAjaW5jbHVkZSA8bm1taW50cmluLmg+XG4vLyNlbmRpZlxuXG5cblxuc3RydWN0IFNlYXJjaFN5bWJvbHNcbntcbiAgICBzdGF0aWMgY29uc3RleHByIGF1dG8gQlVGRkVSX1NJWkUgPSAxNjtcblxuICAgIFNlYXJjaFN5bWJvbHMoKSA9IGRlZmF1bHQ7XG5cbiAgICBTZWFyY2hTeW1ib2xzKHN0ZDo6c3RyaW5nX3ZpZXcgaW4pXG4gICAgICAgIDogc3RyKGluLnN1YnN0cigwLCBzdGQ6Om1pbjxzaXplX3Q+KEJVRkZFUl9TSVpFLCBpbi5zaXplKCkpKSlcbiAgICB7XG4vLyNpZiBkZWZpbmVkKF9fU1NFNF8yX18pXG4vLyAgICAgICAgaWYgKHN0ci5zaXplKCkgPiBCVUZGRVJfU0laRSlcbi8vICAgICAgICB7XG4vLyAgICAgICAgICAgIHRocm93IHN0ZDo6cnVudGltZV9lcnJvcihcIlNlYXJjaFN5bWJvbHMgY2FuIGNvbnRhaW4gYXQgbW9zdCBcIiArIHN0ZDo6dG9fc3RyaW5nKEJVRkZFUl9TSVpFKSArIFwiIHN5bWJvbHMgYW5kIFwiICsgc3RkOjp0b19zdHJpbmcoc3RyLnNpemUoKSkgKyBcIiB3YXMgcHJvdmlkZWRcXG5cIik7XG4vLyAgICAgICAgfVxuXG4gICAgICAgIGNoYXIgdG1wX3NhZmV0eV9idWZmZXJbQlVGRkVSX1NJWkVdID0gezB9O1xuXG4gICAgICAgIG1lbWNweSh0bXBfc2FmZXR5X2J1ZmZlciwgc3RyLmRhdGEoKSwgc3RyLnNpemUoKSk7XG5cbiAgICAgICAgc2ltZF92ZWN0b3IgPSBfbW1fbG9hZHVfc2kxMjgocmVpbnRlcnByZXRfY2FzdDxjb25zdCBfX20xMjhpICo+KHRtcF9zYWZldHlfYnVmZmVyKSk7XG4vLyNlbmRpZlxuICAgIH1cblxuLy8jaWYgZGVmaW5lZChfX1NTRTRfMl9fKVxuICAgIF9fbTEyOGkgc2ltZF92ZWN0b3I7XG4vLyNlbmRpZlxuICAgIHN0ZDo6c3RyaW5nX3ZpZXcgc3RyO1xufTtcblxubmFtZXNwYWNlIGRldGFpbFxue1xudGVtcGxhdGUgPGNoYXIgLi4uY2hhcnM+IGNvbnN0ZXhwciBib29sIGlzX2luKGNoYXIgeCkgeyByZXR1cm4gKCh4ID09IGNoYXJzKSB8fCAuLi4pOyB9XG5cbnN0YXRpYyBib29sIGlzX2luKGNoYXIgYywgY29uc3QgY2hhciAqIHN5bWJvbHMsIHNpemVfdCBudW1fY2hhcnMpXG57XG4gICAgZm9yIChzaXplX3QgaSA9IDB1OyBpIDwgbnVtX2NoYXJzOyArK2kpXG4gICAge1xuICAgICAgICBpZiAoYyA9PSBzeW1ib2xzW2ldKVxuICAgICAgICB7XG4gICAgICAgICAgICByZXR1cm4gdHJ1ZTtcbiAgICAgICAgfVxuICAgIH1cblxuICAgIHJldHVybiBmYWxzZTtcbn1cblxuaW5saW5lIF9fbTEyOGkgbW1faXNfaW4oX19tMTI4aSBieXRlcywgY29uc3QgY2hhciAqIHN5bWJvbHMsIHNpemVfdCBudW1fY2hhcnMpXG57XG4gICAgX19tMTI4aSBhY2N1bXVsYXRvciA9IF9tbV9zZXR6ZXJvX3NpMTI4KCk7XG4gICAgZm9yIChzaXplX3QgaSA9IDA7IGkgPCBudW1fY2hhcnM7ICsraSlcbiAgICB7XG4gICAgICAgIF9fbTEyOGkgZXEgPSBfbW1fY21wZXFfZXBpOChieXRlcywgX21tX3NldDFfZXBpOChzeW1ib2xzW2ldKSk7XG4gICAgICAgIGFjY3VtdWxhdG9yID0gX21tX29yX3NpMTI4KGFjY3VtdWxhdG9yLCBlcSk7XG4gICAgfVxuXG4gICAgcmV0dXJuIGFjY3VtdWxhdG9yO1xufVxuXG5pbmxpbmUgc3RkOjphcnJheTxfX20xMjhpLCAxNnU+IG1tX2lzX2luX3ByZXBhcmUoY29uc3QgY2hhciAqIHN5bWJvbHMsIHNpemVfdCBudW1fY2hhcnMpXG57XG4gICAgc3RkOjphcnJheTxfX20xMjhpLCAxNnU+IHJlc3VsdCB7fTtcblxuICAgIGZvciAoc2l6ZV90IGkgPSAwOyBpIDwgbnVtX2NoYXJzOyArK2kpXG4gICAge1xuICAgICAgICByZXN1bHRbaV0gPSBfbW1fc2V0MV9lcGk4KHN5bWJvbHNbaV0pO1xuICAgIH1cblxuICAgIHJldHVybiByZXN1bHQ7XG59XG5cbmlubGluZSBfX20xMjhpIG1tX2lzX2luX2V4ZWN1dGUoX19tMTI4aSBieXRlcywgY29uc3Qgc3RkOjphcnJheTxfX20xMjhpLCAxNnU+ICYgbmVlZGxlcylcbntcbiAgICBfX20xMjhpIGFjY3VtdWxhdG9yID0gX21tX3NldHplcm9fc2kxMjgoKTtcblxuICAgIGZvciAoY29uc3QgYXV0byAmIG5lZWRsZSA6IG5lZWRsZXMpXG4gICAge1xuICAgICAgICBfX20xMjhpIGVxID0gX21tX2NtcGVxX2VwaTgoYnl0ZXMsIG5lZWRsZSk7XG4gICAgICAgIGFjY3VtdWxhdG9yID0gX21tX29yX3NpMTI4KGFjY3VtdWxhdG9yLCBlcSk7XG4gICAgfVxuXG4gICAgcmV0dXJuIGFjY3VtdWxhdG9yO1xufVxuXG50ZW1wbGF0ZSA8Ym9vbCBwb3NpdGl2ZT5cbmNvbnN0ZXhwciBib29sIG1heWJlX25lZ2F0ZShib29sIHgpIHsgcmV0dXJuIHggPT0gcG9zaXRpdmU7IH1cblxudGVtcGxhdGUgPGJvb2wgcG9zaXRpdmU+XG5jb25zdGV4cHIgdWludDE2X3QgbWF5YmVfbmVnYXRlKHVpbnQxNl90IHgpXG57XG4gICAgaWYgY29uc3RleHByIChwb3NpdGl2ZSlcbiAgICAgICAgcmV0dXJuIHg7XG4gICAgZWxzZVxuICAgICAgICByZXR1cm4gfng7XG59XG5cbmVudW0gY2xhc3MgUmV0dXJuTW9kZVxue1xuICAgIEVuZCxcbiAgICBOdWxscHRyLFxufTtcblxuXG50ZW1wbGF0ZSA8Ym9vbCBwb3NpdGl2ZSwgUmV0dXJuTW9kZSByZXR1cm5fbW9kZT5cbmlubGluZSBjb25zdCBjaGFyICogZmluZF9maXJzdF9zeW1ib2xzX3NzZTIoY29uc3QgY2hhciAqIGNvbnN0IGJlZ2luLCBjb25zdCBjaGFyICogY29uc3QgZW5kLCBjb25zdCBjaGFyICogc3ltYm9scywgc2l6ZV90IG51bV9jaGFycylcbntcbiAgICBjb25zdCBjaGFyICogcG9zID0gYmVnaW47XG5cbiNpZiBkZWZpbmVkKF9fU1NFMl9fKVxuICAgIGNvbnN0IGF1dG8gbmVlZGxlcyA9IG1tX2lzX2luX3ByZXBhcmUoc3ltYm9scywgbnVtX2NoYXJzKTtcbiAgICBmb3IgKDsgcG9zICsgMTUgPCBlbmQ7IHBvcyArPSAxNilcbiAgICB7XG4gICAgICAgIF9fbTEyOGkgYnl0ZXMgPSBfbW1fbG9hZHVfc2kxMjgocmVpbnRlcnByZXRfY2FzdDxjb25zdCBfX20xMjhpICo+KHBvcykpO1xuXG4gICAgICAgIF9fbTEyOGkgZXEgPSBtbV9pc19pbl9leGVjdXRlKGJ5dGVzLCBuZWVkbGVzKTtcblxuICAgICAgICB1aW50MTZfdCBiaXRfbWFzayA9IG1heWJlX25lZ2F0ZTxwb3NpdGl2ZT4odWludDE2X3QoX21tX21vdmVtYXNrX2VwaTgoZXEpKSk7XG4gICAgICAgIGlmIChiaXRfbWFzaylcbiAgICAgICAgICAgIHJldHVybiBwb3MgKyBfX2J1aWx0aW5fY3R6KGJpdF9tYXNrKTtcbiAgICB9XG4jZW5kaWZcblxuICAgIGZvciAoOyBwb3MgPCBlbmQ7ICsrcG9zKVxuICAgICAgICBpZiAobWF5YmVfbmVnYXRlPHBvc2l0aXZlPihpc19pbigqcG9zLCBzeW1ib2xzLCBudW1fY2hhcnMpKSlcbiAgICAgICAgICAgIHJldHVybiBwb3M7XG5cbiAgICByZXR1cm4gcmV0dXJuX21vZGUgPT0gUmV0dXJuTW9kZTo6RW5kID8gZW5kIDogbnVsbHB0cjtcbn1cblxuXG50ZW1wbGF0ZSA8Ym9vbCBwb3NpdGl2ZSwgUmV0dXJuTW9kZSByZXR1cm5fbW9kZT5cbmlubGluZSBjb25zdCBjaGFyICogZmluZF9maXJzdF9zeW1ib2xzX3NzZTQyKGNvbnN0IGNoYXIgKiBjb25zdCBiZWdpbiwgY29uc3QgY2hhciAqIGNvbnN0IGVuZCwgY29uc3QgU2VhcmNoU3ltYm9scyAmIHN5bWJvbHMpXG57XG4gICAgY29uc3QgY2hhciAqIHBvcyA9IGJlZ2luO1xuXG4gICAgY29uc3QgYXV0byBudW1fY2hhcnMgPSBzeW1ib2xzLnN0ci5zaXplKCk7XG5cbi8vI2lmIGRlZmluZWQoX19TU0U0XzJfXylcbiAgICBjb25zdGV4cHIgaW50IG1vZGUgPSBfU0lERF9VQllURV9PUFMgfCBfU0lERF9DTVBfRVFVQUxfQU5ZIHwgX1NJRERfTEVBU1RfU0lHTklGSUNBTlQ7XG5cbiAgICBjb25zdCBfX20xMjhpIHNldCA9IHN5bWJvbHMuc2ltZF92ZWN0b3I7XG5cbiAgICBmb3IgKDsgcG9zICsgMTUgPCBlbmQ7IHBvcyArPSAxNilcbiAgICB7XG4gICAgICAgIF9fbTEyOGkgYnl0ZXMgPSBfbW1fbG9hZHVfc2kxMjgocmVpbnRlcnByZXRfY2FzdDxjb25zdCBfX20xMjhpICo+KHBvcykpO1xuXG4gICAgICAgIGlmIGNvbnN0ZXhwciAocG9zaXRpdmUpXG4gICAgICAgIHtcbiAgICAgICAgICAgIGlmIChfbW1fY21wZXN0cmMoc2V0LCBudW1fY2hhcnMsIGJ5dGVzLCAxNiwgbW9kZSkpXG4gICAgICAgICAgICAgICAgcmV0dXJuIHBvcyArIF9tbV9jbXBlc3RyaShzZXQsIG51bV9jaGFycywgYnl0ZXMsIDE2LCBtb2RlKTtcbiAgICAgICAgfVxuICAgICAgICBlbHNlXG4gICAgICAgIHtcbiAgICAgICAgICAgIGlmIChfbW1fY21wZXN0cmMoc2V0LCBudW1fY2hhcnMsIGJ5dGVzLCAxNiwgbW9kZSB8IF9TSUREX05FR0FUSVZFX1BPTEFSSVRZKSlcbiAgICAgICAgICAgICAgICByZXR1cm4gcG9zICsgX21tX2NtcGVzdHJpKHNldCwgbnVtX2NoYXJzLCBieXRlcywgMTYsIG1vZGUgfCBfU0lERF9ORUdBVElWRV9QT0xBUklUWSk7XG4gICAgICAgIH1cbiAgICB9XG4vLyNlbmRpZlxuXG4gICAgZm9yICg7IHBvcyA8IGVuZDsgKytwb3MpXG4gICAgICAgIGlmIChtYXliZV9uZWdhdGU8cG9zaXRpdmU+KGlzX2luKCpwb3MsIHN5bWJvbHMuc3RyLmRhdGEoKSwgbnVtX2NoYXJzKSkpXG4gICAgICAgICAgICByZXR1cm4gcG9zO1xuXG4gICAgcmV0dXJuIHJldHVybl9tb2RlID09IFJldHVybk1vZGU6OkVuZCA/IGVuZCA6IG51bGxwdHI7XG59XG5cbi8vLyBOT1RFIE5vIFNTRSA0LjIgaW1wbGVtZW50YXRpb24gZm9yIGZpbmRfbGFzdF9zeW1ib2xzX29yX251bGwuIE5vdCB3b3J0aCB0byBkby5cblxuXG50ZW1wbGF0ZSA8Ym9vbCBwb3NpdGl2ZSwgUmV0dXJuTW9kZSByZXR1cm5fbW9kZT5cbmlubGluZSBjb25zdCBjaGFyICogZmluZF9maXJzdF9zeW1ib2xzX2Rpc3BhdGNoKGNvbnN0IHN0ZDo6c3RyaW5nX3ZpZXcgaGF5c3RhY2ssIGNvbnN0IFNlYXJjaFN5bWJvbHMgJiBzeW1ib2xzKVxue1xuLy8jaWYgZGVmaW5lZChfX1NTRTRfMl9fKVxuICAgIGlmIChzeW1ib2xzLnN0ci5zaXplKCkgPj0gNSlcbiAgICAgICAgcmV0dXJuIGZpbmRfZmlyc3Rfc3ltYm9sc19zc2U0Mjxwb3NpdGl2ZSwgcmV0dXJuX21vZGU+KGhheXN0YWNrLmJlZ2luKCksIGhheXN0YWNrLmVuZCgpLCBzeW1ib2xzKTtcbiAgICBlbHNlXG4vLyNlbmRpZlxuICAgICAgICByZXR1cm4gZmluZF9maXJzdF9zeW1ib2xzX3NzZTI8cG9zaXRpdmUsIHJldHVybl9tb2RlPihoYXlzdGFjay5iZWdpbigpLCBoYXlzdGFjay5lbmQoKSwgc3ltYm9scy5zdHIuZGF0YSgpLCBzeW1ib2xzLnN0ci5zaXplKCkpO1xufVxuXG59XG5cblxuXG5pbmxpbmUgY29uc3QgY2hhciAqIGZpbmRfZmlyc3Rfc3ltYm9scyhzdGQ6OnN0cmluZ192aWV3IGhheXN0YWNrLCBjb25zdCBTZWFyY2hTeW1ib2xzICYgc3ltYm9scylcbntcbiAgICByZXR1cm4gZGV0YWlsOjpmaW5kX2ZpcnN0X3N5bWJvbHNfZGlzcGF0Y2g8dHJ1ZSwgZGV0YWlsOjpSZXR1cm5Nb2RlOjpFbmQ+KGhheXN0YWNrLCBzeW1ib2xzKTtcbn1cblxuXG5cbmlubGluZSBjb25zdCBjaGFyICogZmluZF9maXJzdF9ub3Rfc3ltYm9scyhzdGQ6OnN0cmluZ192aWV3IGhheXN0YWNrLCBjb25zdCBTZWFyY2hTeW1ib2xzICYgc3ltYm9scylcbntcbiAgICByZXR1cm4gZGV0YWlsOjpmaW5kX2ZpcnN0X3N5bWJvbHNfZGlzcGF0Y2g8ZmFsc2UsIGRldGFpbDo6UmV0dXJuTW9kZTo6RW5kPihoYXlzdGFjaywgc3ltYm9scyk7XG59XG5cbmlubGluZSBjb25zdCBjaGFyICogZmluZF9maXJzdF9zeW1ib2xzX29yX251bGwoc3RkOjpzdHJpbmdfdmlldyBoYXlzdGFjaywgY29uc3QgU2VhcmNoU3ltYm9scyAmIHN5bWJvbHMpXG57XG4gICAgcmV0dXJuIGRldGFpbDo6ZmluZF9maXJzdF9zeW1ib2xzX2Rpc3BhdGNoPHRydWUsIGRldGFpbDo6UmV0dXJuTW9kZTo6TnVsbHB0cj4oaGF5c3RhY2ssIHN5bWJvbHMpO1xufVxuXG5cbmlubGluZSBjb25zdCBjaGFyICogZmluZF9maXJzdF9ub3Rfc3ltYm9sc19vcl9udWxsKHN0ZDo6c3RyaW5nX3ZpZXcgaGF5c3RhY2ssIGNvbnN0IFNlYXJjaFN5bWJvbHMgJiBzeW1ib2xzKVxue1xuICAgIHJldHVybiBkZXRhaWw6OmZpbmRfZmlyc3Rfc3ltYm9sc19kaXNwYXRjaDxmYWxzZSwgZGV0YWlsOjpSZXR1cm5Nb2RlOjpOdWxscHRyPihoYXlzdGFjaywgc3ltYm9scyk7XG59XG5cblxudXNpbmcgbmFtZXNwYWNlIHN0ZDo6bGl0ZXJhbHM7XG5cbnN0YXRpYyBjb25zdCBhdXRvIGhheXN0YWNrID0gUlwidGV4dChcbjEpIFRoZSBleGVjdXRpb24gcG9saWN5IHR5cGUgdXNlZCBhcyBhIHVuaXF1ZSB0eXBlIHRvIGRpc2FtYmlndWF0ZSBwYXJhbGxlbCBhbGdvcml0aG0gb3ZlcmxvYWRpbmcgYW5kIHJlcXVpcmUgdGhhdCBhIHBhcmFsbGVsIGFsZ29yaXRobSdzIGV4ZWN1dGlvbiBtYXkgbm90IGJlIHBhcmFsbGVsaXplZC4gVGhlIGludm9jYXRpb25zIG9mIGVsZW1lbnQgYWNjZXNzIGZ1bmN0aW9ucyBpbiBwYXJhbGxlbCBhbGdvcml0aG1zIGludm9rZWQgd2l0aCB0aGlzIHBvbGljeSAodXN1YWxseSBzcGVjaWZpZWQgYXMgc3RkOjpleGVjdXRpb246OnNlcSkgYXJlIGluZGV0ZXJtaW5hdGVseSBzZXF1ZW5jZWQgaW4gdGhlIGNhbGxpbmcgdGhyZWFkLlxuMikgVGhlIGV4ZWN1dGlvbiBwb2xpY3kgdHlwZSB1c2VkIGFzIGEgdW5pcXVlIHR5cGUgdG8gZGlzYW1iaWd1YXRlIHBhcmFsbGVsIGFsZ29yaXRobSBvdmVybG9hZGluZyBhbmQgaW5kaWNhdGUgdGhhdCBhIHBhcmFsbGVsIGFsZ29yaXRobSdzIGV4ZWN1dGlvbiBtYXkgYmUgcGFyYWxsZWxpemVkLiBUaGUgaW52b2NhdGlvbnMgb2YgZWxlbWVudCBhY2Nlc3MgZnVuY3Rpb25zIGluIHBhcmFsbGVsIGFsZ29yaXRobXMgaW52b2tlZCB3aXRoIHRoaXMgcG9saWN5ICh1c3VhbGx5IHNwZWNpZmllZCBhcyBzdGQ6OmV4ZWN1dGlvbjo6cGFyKSBhcmUgcGVybWl0dGVkIHRvIGV4ZWN1dGUgaW4gZWl0aGVyIHRoZSBpbnZva2luZyB0aHJlYWQgb3IgaW4gYSB0aHJlYWQgaW1wbGljaXRseSBjcmVhdGVkIGJ5IHRoZSBsaWJyYXJ5IHRvIHN1cHBvcnQgcGFyYWxsZWwgYWxnb3JpdGhtIGV4ZWN1dGlvbi4gQW55IHN1Y2ggaW52b2NhdGlvbnMgZXhlY3V0aW5nIGluIHRoZSBzYW1lIHRocmVhZCBhcmUgaW5kZXRlcm1pbmF0ZWx5IHNlcXVlbmNlZCB3aXRoIHJlc3BlY3QgdG8gZWFjaCBvdGhlci5cbjMpIFRoZSBleGVjdXRpb24gcG9saWN5IHR5cGUgdXNlZCBhcyBhIHVuaXF1ZSB0eXBlIHRvIGRpc2FtYmlndWF0ZSBwYXJhbGxlbCBhbGdvcml0aG0gb3ZlcmxvYWRpbmcgYW5kIGluZGljYXRlIHRoYXQgYSBwYXJhbGxlbCBhbGdvcml0aG0ncyBleGVjdXRpb24gbWF5IGJlIHBhcmFsbGVsaXplZCwgdmVjdG9yaXplZCwgb3IgbWlncmF0ZWQgYWNyb3NzIHRocmVhZHMgKHN1Y2ggYXMgYnkgYSBwYXJlbnQtc3RlYWxpbmcgc2NoZWR1bGVyKS4gVGhlIGludm9jYXRpb25zIG9mIGVsZW1lbnQgYWNjZXNzIGZ1bmN0aW9ucyBpbiBwYXJhbGxlbCBhbGdvcml0aG1zIGludm9rZWQgd2l0aCB0aGlzIHBvbGljeSBhcmUgcGVybWl0dGVkIHRvIGV4ZWN1dGUgaW4gYW4gdW5vcmRlcmVkIGZhc2hpb24gaW4gdW5zcGVjaWZpZWQgdGhyZWFkcywgYW5kIHVuc2VxdWVuY2VkIHdpdGggcmVzcGVjdCB0byBvbmUgYW5vdGhlciB3aXRoaW4gZWFjaCB0aHJlYWQuXG40KSBUaGUgZXhlY3V0aW9uIHBvbGljeSB0eXBlIHVzZWQgYXMgYSB1bmlxdWUgdHlwZSB0byBkaXNhbWJpZ3VhdGUgcGFyYWxsZWwgYWxnb3JpdGhtIG92ZXJsb2FkaW5nIGFuZCBpbmRpY2F0ZSB0aGF0IGEgcGFyYWxsZWwgYWxnb3JpdGhtJ3MgZXhlY3V0aW9uIG1heSBiZSB2ZWN0b3JpemVkLCBlLmcuLCBleGVjdXRlZCBvbiBhIHNpbmdsZSB0aHJlYWQgdXNpbmcgaW5zdHJ1Y3Rpb25zIHRoYXQgb3BlcmF0ZSBvbiBtdWx0aXBsZSBkYXRhIGl0ZW1zLlxufiFAIyQlXiYqKCkvXG4pdGV4dFwic3Y7XG5cbi8vIEhlcmUgd2UgbWFrZSBzdXJlIG5lZWRsZXMgYXJlIGF0IGxlYXN0IDE2IGJ5dGVzIGxvbmcsIHdlJ2xsIGNyZWF0ZSBzaG9ydGVyIG5lZWRsZXMgZnJvbSB0aGF0IGxhdGVyXG5zdGF0aWMgY29uc3Qgc3RkOjpzdHJpbmdfdmlldyB0ZXN0X25lZWRsZXNbXSA9IHtcbiAgICBcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcImBfXFxcIlwic3YsIC8vIGRvZXNuJ3QgbWF0Y2hcbiAgICBoYXlzdGFjay5zdWJzdHIoaGF5c3RhY2suc2l6ZSgpIC0gNjQsIGhheXN0YWNrLnNpemUoKSksXG4gICAgaGF5c3RhY2suc3Vic3RyKGhheXN0YWNrLnNpemUoKSAtIDMyLCBoYXlzdGFjay5zaXplKCkpLFxuICAgIGhheXN0YWNrLnN1YnN0cihoYXlzdGFjay5zaXplKCkgLSAxNiwgaGF5c3RhY2suc2l6ZSgpKSxcbn07XG5cbmF1dG8gbWF4X3NpemVfbmVlZGxlcyhjb25zdCBhdXRvICYgaW5fbmVlZGxlcywgc2l6ZV90IG1heF9zaXplKSB7XG4gICAgc3RkOjp2ZWN0b3I8c3RkOjpkZWNheV90PGRlY2x0eXBlKGluX25lZWRsZXNbMF0pPj4gcmVzdWx0O1xuICAgIHJlc3VsdC5yZXNpemUoc3RkOjpzaXplKGluX25lZWRsZXMpKTtcblxuICAgIGZvciAoY29uc3QgYXV0byAmIG4gOiBpbl9uZWVkbGVzKSB7XG4gICAgICAgIHJlc3VsdC5wdXNoX2JhY2sobi5zdWJzdHIoc3RkOjptaW48c2l6ZV90PihuLnNpemUoKSwgbWF4X3NpemUpKSk7XG4gICAgfVxuXG4gICAgcmV0dXJuIHJlc3VsdDtcbn1cblxuc3RhdGljIHZvaWQgQk1fZmZzX1NUTChiZW5jaG1hcms6OlN0YXRlJiBzdGF0ZSkge1xuXG4gICAgY29uc3QgYXV0byBuZWVkbGVzID0gbWF4X3NpemVfbmVlZGxlcyh0ZXN0X25lZWRsZXMsIHN0YXRlLnJhbmdlKDApKTtcblxuICAgIGZvciAoYXV0byBfIDogc3RhdGUpIHtcbiAgICAgICAgZm9yIChjb25zdCBhdXRvICYgbmVlZGxlIDogbmVlZGxlcykge1xuICAgICAgICAgICAgYmVuY2htYXJrOjpEb05vdE9wdGltaXplKFxuICAgICAgICAgICAgICAgIHN0ZDo6ZmluZF9maXJzdF9vZihzdGQ6OmV4ZWN1dGlvbjo6dW5zZXEsIGhheXN0YWNrLmJlZ2luKCksIGhheXN0YWNrLmVuZCgpLCBuZWVkbGUuYmVnaW4oKSwgbmVlZGxlLmVuZCgpKVxuICAgICAgICAgICAgKTtcbiAgICAgICAgfVxuICAgIH1cbn1cblxuXG5zdGF0aWMgdm9pZCBCTV9mZnNfU1NFKGJlbmNobWFyazo6U3RhdGUmIHN0YXRlKSB7XG4gIC8vIFBlcmZvcm0gc2V0dXAgaGVyZVxuICAgIGNvbnN0IGF1dG8gbmVlZGxlcyA9IG1heF9zaXplX25lZWRsZXModGVzdF9uZWVkbGVzLCBzdGF0ZS5yYW5nZSgwKSk7XG5cbiAgICBzdGQ6OnZlY3RvcjxTZWFyY2hTeW1ib2xzPiBwcmVwYXJlZF9uZWVkZWxlcztcbiAgICBwcmVwYXJlZF9uZWVkZWxlcy5yZXNlcnZlKHN0ZDo6c2l6ZShuZWVkbGVzKSk7XG4gICAgZm9yIChjb25zdCBhdXRvICYgbmVlZGxlIDogbmVlZGxlcylcbiAgICAgICAgcHJlcGFyZWRfbmVlZGVsZXMuZW1wbGFjZV9iYWNrKG5lZWRsZSk7XG5cbiAgICBmb3IgKGF1dG8gXyA6IHN0YXRlKSB7XG4gICAgICAgIGZvciAoY29uc3QgYXV0byAmIG5lZWRsZSA6IHByZXBhcmVkX25lZWRlbGVzKVxuICAgICAgICAgICAgYmVuY2htYXJrOjpEb05vdE9wdGltaXplKFxuICAgICAgICAgICAgICAgIGZpbmRfZmlyc3Rfc3ltYm9sc19vcl9udWxsKGhheXN0YWNrLCBuZWVkbGUpXG4gICAgICAgICAgICApO1xuICAgIH1cbn1cblxuQkVOQ0hNQVJLKEJNX2Zmc19TVEwpLT5BcmcoMiktPkFyZyg0KS0+QXJnKDgpLT5BcmcoMTYpO1xuQkVOQ0hNQVJLKEJNX2Zmc19TU0UpLT5BcmcoMiktPkFyZyg0KS0+QXJnKDgpLT5BcmcoMTYpO1xuXG5cbiNwcmFnbWEgY2xhbmcgYXR0cmlidXRlIHBvcCIsImNvbXBpbGVyIjoiY2xhbmctMTQuMCIsIm9wdGltIjoiMiJ9) `find_first_symbols_or_null` is significantly faster than corresponding `std::find_first_of(std::execution::unseq, ...`

![image](https://user-images.githubusercontent.com/1110183/230782111-451be529-24b1-4dc9-853b-fc2fd3f0d0d0.png)